### ISSUE TITLE
EU managed cloud

### ISSUE THREAD

**Issue by jamilbk** [CloudFire-LLC/cloudfire#3155]
For customers with strict compliance requirements that are based in the EU, it may be helpful if the Firezone managed control plane supported an EU-locked cloud offering to restrict data persistence to GCP EU cloud regions only.

**Comment by AndrewDryga** [comment]
I think for some of them it will be fine to have one portal but restrict the data routing just to EU or self-hosted relays? Because the portal doesn't store they company data other than synced users, which should be fine considering most clouds do the same thing even if they provide region-local VMs.

**Comment by jamilbk** [comment]
Nah I made sure to clarify this particular point. Cloud providers have jumped through many more legal hoops than we have or will any time soon, so it's a different story for them.

This issue would be particularly for the control plane.

**Comment by jamilbk** [comment]
Had another customer (healthcare, EU) where this is a tablestakes requirement.

**Comment by small1** [comment]
A self hosted control plane would be the most optimal way to go.  Because it is not sufficient with just something eu managed cloud.  The entity managing it needs to be eu based as well and not US an us owned entity.

**Comment by AndrewDryga** [comment]
> A self hosted control plane would be the most optimal way to go.  Because it is not sufficient with just something eu managed cloud.  The entity managing it needs to be eu based as well and not US an us owned entity.

Can you please refer to a regulation that prohibits the use of software that is run by US-based entities in Europe? Pretty much all the laws Im aware of just regulate data governance, not where the company is registered.

**Comment by small1** [comment]
GDPR vs FISA 702.  Personal information can be fetched from any us owned entity regardless of where servers are located.  It is not enough to just select an azure server in Germany for example aince the server itself is run by Microsoft.

**Comment by AndrewDryga** [comment]
@small1 thank you. Do I understand correctly that there is no regulation but just a chance that US government will want to request your company data and we will have to comply? 

Where you would host it to protect from it, some local data center with OpenStack or something like that? If one day it goes down, what kind of support you expect from self-hosted version? How do you want to store backups?

Also, just wanted to point our that our data is end-to-end encrypted with WireGuard, so the only data that we actually store are OIDC access tokens with limited scope, directory (email, group and name) and authorization logs (who accessed what and when). We won't be able to provide any actual data flowing through the tunnel even if we really wanted to.

As one of features for the future we were also thinking to allow users to use their own PKI so clients and gateways will use provided keys for cryptography.

**Comment by small1** [comment]
Yes you would have to comply and it would probably come with an NDA sumo that you can't inform customer/customers about it.

Inlines the simplicity of the 0.7 ersion with just a docker container with everything in.  But for some deployments it would be on openstack.  On some others it would be on proxmox.  For myself I have quite decent backup infrastructure even for home use. 

On self hosted I would not expect any support except forums.  But a self hosted enterprise (I have two that might have use for that) would be good.  But I would also expect if one has a complete Openstack infra with redundancy they would know a bit about running complex setups.

**Comment by AndrewDryga** [comment]
Well, if we talk about home use you can totally do it today even though we don't recommend it due to involved complexity and because if it breaks then you are on your own. 

You can take inspiration from Terraform definitions that we use in production and our docker-compose file that is used for local development to build your own setup. But you will have to build your own apps, etc.
